runTests.py fails on CygWin

#### Summary:
Executing tests via runTests.py fails on my dev machine (Windows 10 + CygWin).

#### Description:
When using the CygWin terminal on Windows 10, runTests.py does not correctly detect that the machine is not windows and does not add .exe to target names (see #338). The problem is that on my system (Windows 10 + CygWin + Python 2.7) a call to ```platform.system()``` returns "CYGWIN_NT-10.0" while the ```isWin()``` function in runTests.py only checks for ```Windows```.


#### Reproducible Steps:
On CygWin, using the CygWin terminal, run 
```
python runTests.py src/test
```

#### Current Output:

```
--- lots of stuff ---
test/integration/compile_models.o:compile_models_test.cpp:(.rdata$_ZTV24lang_copile_models_Test[_ZTV24lang_compile_models_Test]+0x20): undefined reference to testing::Test::SetUp()'
test/integration/compile_models.o:compile_models_test.cpp:(.rdata$_ZTV24lang_copile_models_Test[_ZTV24lang_compile_models_Test]+0x28): undefined reference to testing::Test::TearDown()'
/usr/lib/gcc/x86_64-pc-cygwin/5.4.0/../../../../lib/libcygwin.a(libcmain.o): Infunction `main':
/usr/src/debug/cygwin-2.6.1-1/winsup/cygwin/lib/libcmain.c:37: undefined referece to `WinMain'
collect2: error: ld returned 1 exit status
make: *** [test/integration/compile_models] Error 1
rm test/integration/compile_models.o
make -j4 test/integration/compile_models failed
exit now (04/10/17 11:15:01 CEST)
```


#### Expected Output:
Compile the tests and run them.


#### Additional Information:
Modifying the check in ```isWin()``` makes the problem go away (and I run into #2058, but that's another story).

#### Current Version:
v2.14.0

--- a/runTests.py
+++ b/runTests.py
@@ -30,6 +30,7 @@ def stopErr(msg, returncode):
 
 def isWin():
     if (platform.system().lower().startswith("windows")
+        or platform.system().lower().startswith("cygwin")
         or os.name.lower().startswith("windows")):
         return True
     return False

